opensearch-operator/pkg/builders/cluster.go

```diff
@@ -971,3 +972,81 @@ func DataNodesCount(ctx context.Context, k8sClient client.Client, cr *opsterv1.O
 	}
 	return count
 }
+
+func NewServiceMonitor(cr *opsterv1.OpenSearchCluster) *monitoring.ServiceMonitor {
+	ClusterLabels := "opster.io/opensearch-cluster"
+
+	labels := map[string]string{
+		ClusterLabels: cr.Name,
+	}
+	selector := metav1.LabelSelector{
+		MatchLabels: labels,
+	}
+
+	namespaceSelector := monitoring.NamespaceSelector{
+		Any:        false,
+		MatchNames: []string{cr.Namespace},
+	}
+
+	if cr.Spec.General.Monitoring.ScrapeInterval == "" {
+		cr.Spec.General.Monitoring.ScrapeInterval = "30s"
+	}
+	user := monitoring.BasicAuth{}
+
+	monitorUser := cr.Spec.General.Monitoring.MonitoringUserSecret
+	if monitorUser == "" {
+		// Use admin credentials if no separate monitoring user was defined
+		user = monitoring.BasicAuth{
+			Username: corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{Name: cr.Name + "-admin-password"},
+				Key:                  "username",
+			},
+			Password: corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{Name: cr.Name + "-admin-password"},
+				Key:                  "password",
+			},
+		}
+	} else {
+		user = monitoring.BasicAuth{
+			Username: corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{Name: cr.Spec.General.Monitoring.MonitoringUserSecret},
+				Key:                  "username",
+			},
+			Password: corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{Name: cr.Spec.General.Monitoring.MonitoringUserSecret},
+				Key:                  "password",
+			},
+		}
+
+	}
+
+	return &monitoring.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name + "-monitor",
+			Namespace: cr.Namespace,
+			Labels:    labels,
+		},
+		Spec: monitoring.ServiceMonitorSpec{
+			JobLabel: cr.Name + "-monitor",
+			TargetLabels: []string{
+				cr.Name,
```

Comment:
Shouldn't this be `labels`?

Reply:
So the service monitor struct dont have `labels` filed, only that `TargetLabels` and `PodTargetLabels`

Reply:
Read the docs again, this should be `helpers.ClusterLabels`, as you specify here the names of labels that should be part of the metric. `cr.Name` would be the value, but you need to give the key.
Same applies for `PodTargetLabels`.